{"ast":null,"code":"import _classCallCheck from \"@babel/runtime/helpers/classCallCheck\";\nimport _createClass from \"@babel/runtime/helpers/createClass\";\nimport _possibleConstructorReturn from \"@babel/runtime/helpers/possibleConstructorReturn\";\nimport _getPrototypeOf from \"@babel/runtime/helpers/getPrototypeOf\";\nimport _inherits from \"@babel/runtime/helpers/inherits\";\nvar _jsxFileName = \"/home/op-pro/Desktop/Finalyear project/Admin/node_modules/react-native-error-boundary/src/ErrorBoundary/index.js\";\nimport React from 'react';\nimport FallbackComponent from \"./FallbackComponent\";\n\nvar ErrorBoundary = function (_React$Component) {\n  _inherits(ErrorBoundary, _React$Component);\n\n  function ErrorBoundary() {\n    var _getPrototypeOf2;\n\n    var _this;\n\n    _classCallCheck(this, ErrorBoundary);\n\n    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {\n      args[_key] = arguments[_key];\n    }\n\n    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(ErrorBoundary)).call.apply(_getPrototypeOf2, [this].concat(args)));\n    _this.state = {\n      error: null,\n      hasError: false\n    };\n\n    _this.resetError = function () {\n      _this.setState({\n        error: null,\n        hasError: false\n      });\n    };\n\n    return _this;\n  }\n\n  _createClass(ErrorBoundary, [{\n    key: \"componentDidCatch\",\n    value: function componentDidCatch(error, info) {\n      if (typeof this.props.onError === 'function') {\n        this.props.onError.call(this, error, info.componentStack);\n      }\n    }\n  }, {\n    key: \"render\",\n    value: function render() {\n      var FallbackComponent = this.props.FallbackComponent;\n      return this.state.hasError ? React.createElement(FallbackComponent, {\n        error: this.state.error,\n        resetError: this.resetError,\n        __source: {\n          fileName: _jsxFileName,\n          lineNumber: 39\n        }\n      }) : this.props.children;\n    }\n  }], [{\n    key: \"getDerivedStateFromError\",\n    value: function getDerivedStateFromError(error) {\n      return {\n        error: error,\n        hasError: true\n      };\n    }\n  }]);\n\n  return ErrorBoundary;\n}(React.Component);\n\nErrorBoundary.defaultProps = {\n  FallbackComponent: FallbackComponent\n};\nexport default ErrorBoundary;","map":{"version":3,"sources":["/home/op-pro/Desktop/Finalyear project/Admin/node_modules/react-native-error-boundary/src/ErrorBoundary/index.js"],"names":["React","FallbackComponent","ErrorBoundary","state","error","hasError","resetError","setState","info","props","onError","call","componentStack","children","Component","defaultProps"],"mappings":";;;;;;AACA,OAAOA,KAAP,MAAqD,OAArD;AAEA,OAAOC,iBAAP;;IAUMC,a;;;;;;;;;;;;;;;UACJC,K,GAAQ;AAAEC,MAAAA,KAAK,EAAE,IAAT;AAAeC,MAAAA,QAAQ,EAAE;AAAzB,K;;UAgBRC,U,GAAuB,YAAM;AAC3B,YAAKC,QAAL,CAAc;AAAEH,QAAAA,KAAK,EAAE,IAAT;AAAeC,QAAAA,QAAQ,EAAE;AAAzB,OAAd;AACD,K;;;;;;;sCARkBD,K,EAAcI,I,EAAkC;AACjE,UAAI,OAAO,KAAKC,KAAL,CAAWC,OAAlB,KAA8B,UAAlC,EAA8C;AAC5C,aAAKD,KAAL,CAAWC,OAAX,CAAmBC,IAAnB,CAAwB,IAAxB,EAA8BP,KAA9B,EAAqCI,IAAI,CAACI,cAA1C;AACD;AACF;;;6BAMS;AAAA,UACAX,iBADA,GACsB,KAAKQ,KAD3B,CACAR,iBADA;AAGR,aAAO,KAAKE,KAAL,CAAWE,QAAX,GACH,oBAAC,iBAAD;AACA,QAAA,KAAK,EAAE,KAAKF,KAAL,CAAWC,KADlB;AAEA,QAAA,UAAU,EAAE,KAAKE,UAFjB;AAAA;AAAA;AAAA;AAAA;AAAA,QADG,GAKH,KAAKG,KAAL,CAAWI,QALf;AAMD;;;6CAvBgCT,K,EAAc;AAC7C,aAAO;AAAEA,QAAAA,KAAK,EAALA,KAAF;AAASC,QAAAA,QAAQ,EAAE;AAAnB,OAAP;AACD;;;;EATyBL,KAAK,CAACc,S;;AAA5BZ,a,CAGGa,Y,GAAe;AACpBd,EAAAA,iBAAiB,EAAEA;AADC,C;AA8BxB,eAAeC,aAAf","sourcesContent":["// @flow\nimport React, { type Node, type ComponentType } from 'react'\n\nimport FallbackComponent from './FallbackComponent'\n\ntype Props = {\n  children: Node,\n  FallbackComponent: ComponentType<any>,\n  onError?: Function\n}\n\ntype State = { error: Error | null, hasError: boolean }\n\nclass ErrorBoundary extends React.Component<Props, State> {\n  state = { error: null, hasError: false }\n\n  static defaultProps = {\n    FallbackComponent: FallbackComponent\n  }\n\n  static getDerivedStateFromError (error: Error) {\n    return { error, hasError: true }\n  }\n\n  componentDidCatch (error: Error, info: { componentStack: string }) {\n    if (typeof this.props.onError === 'function') {\n      this.props.onError.call(this, error, info.componentStack)\n    }\n  }\n\n  resetError: Function = () => {\n    this.setState({ error: null, hasError: false })\n  }\n\n  render () {\n    const { FallbackComponent } = this.props\n\n    return this.state.hasError\n      ? <FallbackComponent\n        error={this.state.error}\n        resetError={this.resetError}\n      />\n      : this.props.children\n  }\n}\n\nexport default ErrorBoundary\n"]},"metadata":{},"sourceType":"module"}